util.go

```diff
@@ -38,3 +49,21 @@ func mergeParameters(p1 map[string]Parameter, p2 map[string]Parameter) map[strin
 	}
 	return params
 }
+
+func parseConfig(raw map[string]string, config interface{}) error {
+	dConfig := &mapstructure.DecoderConfig{
```

Comment:
I found [this option](https://github.com/mitchellh/mapstructure/blob/bf980b35cac4dfd34e05254ee5aba086504c3f96/mapstructure.go#L264-L266) - thinking out loud if it would make sense to rename the tag to something more generic like `name`, `config` or maybe even just take the content of the `json` tag? 🤔

Reply:
hmmm `name` wouldn't make sense when you use squash for example ``name:",squash"``, but I think `config` sounds good, donnow about `json` though

Reply:
I see now the linter is complaining because of `json:",squash"`, but now that I think about it I think it would make sense to squash embedded structs by default by setting `Squash` to `true` in `DecoderConfig`. Then you can embed a global config struct in your config as we do in [s3 for example](https://github.com/ConduitIO/conduit-connector-s3/blob/641f6222ec0d191802f5d00f0dff3b66c9dc55de/source/config.go#L35).